File: src/mantle/compat/minecraft/crash/CallableSuppConfig.scala
package mantle.compat.minecraft.crash

import cpw.mods.fml.common.ICrashCallable

/**
 * ICrashCallable for mods to report a supported environment during crashes
 *
 * @param modLabel The name of the mod creating the callable
 * @param message A message to show this is a supported environment e.g. "Environment clean."
 */
class CallableSuppConfig(modLabel:String, message:String) extends ICrashCallable {

  def getLabel = modLabel

  def call = message

}

File: src/mantle/crash/CallableSuppConfig.java
package mantle.crash;

import cpw.mods.fml.common.ICrashCallable;

/**
 * Crash Callable for supported environments.
 *
 * @author Sunstrike <sun@sunstrike.io>
 */
public class CallableSuppConfig implements ICrashCallable
{

    private String modId;

    public CallableSuppConfig(String modId)
    {
        this.modId = modId;
    }

    @Override
    public String call () throws Exception
    {
        return "Environment healthy.";
    }

    @Override
    public String getLabel ()
    {
        return modId + " Environment";
    }

}
